### PR TITLE
#10045 Packing instructions are no longer displayed when ordering 

### PR DESCRIPTION
**Pull request by cristinamghita**
#10045 Packing instructions are no longer displayed when ordering

### CODE REVIEW

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUPIItemProductDAO.java [hunk 1]
```diff
@@ -75,6 +76,9 @@
 
 public class HUPIItemProductDAO implements IHUPIItemProductDAO
 {
+	private static final ISysConfigBL sysConfigBL = Services.get(ISysConfigBL.class);
+	private static final String SYSCONFIG_Enforce_M_HU_PI_Item_Product_ID = "de.metas.handlingunits.impl.HUPIItemProductDAO.Enforce_M_HU_PI_Item_Product_ID";
```
Comment (teosarca) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUPIItemProductDAO.java:80]: move this sysconfig to de.metas.ui.web.quickinput.field.PackingItemProductFieldHelper and call it `de.metas.ui.web.quickinput.field.PackingItemProductFieldHelper.EnforcePrecisePricePerHUItemProduct`.

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUPIItemProductDAO.java [hunk 2]
```diff
@@ -291,10 +295,12 @@ public I_M_HU_PI_Item_Product retrieveMaterialItemProduct(
 			}
 		}
 
+		final boolean enforceHU_PI_Item_Product = sysConfigBL.getBooleanValue(SYSCONFIG_Enforce_M_HU_PI_Item_Product_ID, false);
+		
 		//
 		// Product Price of a Price List Version has this Packing Item
 		{
-			if (queryVO.getPriceListVersionId() != null)
+			if (enforceHU_PI_Item_Product && queryVO.getPriceListVersionId() != null)
```
Comment (teosarca) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUPIItemProductDAO.java:303]: do this checking in de.metas.ui.web.quickinput.field.PackingItemProductFieldHelper and decide there if u set queryVO's priceListVersionId.

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5569170_sys_gh10045_AddSysConfig.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
+-- 2020-10-01T08:04:59.066Z
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_SysConfig (AD_Client_ID,AD_Org_ID,AD_SysConfig_ID,ConfigurationLevel,Created,CreatedBy,EntityType,IsActive,Name,Updated,UpdatedBy,Value) VALUES (0,0,541342,'C',TO_TIMESTAMP('2020-10-01 11:04:58','YYYY-MM-DD HH24:MI:SS'),100,'D','Y','de.metas.handlingunits.impl.HUPIItemProductDAO.Enforce_ M_HU_PI_Item_Product_ID',TO_TIMESTAMP('2020-10-01 11:04:58','YYYY-MM-DD HH24:MI:SS'),100,'N')
```
Comment (teosarca) [inline backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5569170_sys_gh10045_AddSysConfig.sql:3]: rename it to `de.metas.ui.web.quickinput.field.PackingItemProductFieldHelper.EnforcePrecisePricePerHUItemProduct`

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5569170_sys_gh10045_AddSysConfig.sql [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+;
+
+
+-- 2020-10-01T10:34:07.379Z
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+UPDATE AD_SysConfig SET Name='de.metas.handlingunits.impl.HUPIItemProductDAO.Enforce_M_HU_PI_Item_Product_ID',Updated=TO_TIMESTAMP('2020-10-01 13:34:07','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_SysConfig_ID=541342
```
Comment (teosarca) [inline backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5569170_sys_gh10045_AddSysConfig.sql:9]: rename it to `de.metas.ui.web.quickinput.field.PackingItemProductFieldHelper.EnforcePrecisePricePerHUItemProduct`

File: backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/quickinput/field/PackingItemProductFieldHelper.java [hunk 5]
```diff
@@ -88,6 +92,8 @@
 			return Optional.empty();
 		}
 
+		final boolean enforcePrecisePricePerHUItemProduct = sysConfigBL.getBooleanValue(SYSCONFIG_EnforcePrecisePricePerHUItemProduct, false);
```
[the remark this answers is not in the document]
Comment (cristinamghita) [reply backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/quickinput/field/PackingItemProductFieldHelper.java]: Agree. Tested and fixed it! Thx!

**Pull request merged**